client/platform/desktop/backend/native/common.ts

```diff
@@ -496,6 +552,20 @@ async function importMedia(settings: Settings, path: string): Promise<JsonMeta>
   return jsonMeta;
 }
 
+/**
+ * After media conversion we need to remove the transcodingKey to signify it is done
+ */
+async function completeConversion(
+  settings: Settings, datasetId: string, transcodingJobKey: string,
+) {
+  const projectDirInfo = await getValidatedProjectDir(settings, datasetId);
+  const existing = await loadJsonMetadata(projectDirInfo.metaFileAbsPath);
+  if (existing.transcodingJobKey === transcodingJobKey) {
+    existing.transcodingJobKey = undefined;
+  }
+  _saveAsJson(projectDirInfo.metaFileAbsPath, existing);
```

Comment:
move save into conditional? Nothing to save otherwise.

Why would the keys not match?  Should an error be thrown if this unexpected condition occurs?

Also, `_saveAsJson` bypasses lock acquisition, so it should not be used.  Should probably absorb that function into saveMetadata.